### PR TITLE
fix type error on parsing enums

### PR DESCRIPTION
**Pull request by cometkim**
See #5432
See also https://github.com/cometkim/gatsby-plugin-typegen/issues/143

### CODE REVIEW

**Comment by cometkim** [comment]
Maybe It would be better to change the type of enum value to `unknown`. it picks a few possible type errors

```
❯ yarn types:check
yarn run v1.22.5
$ tsc --noEmit
packages/plugins/flow/flow/src/visitor.ts:179:15 - error TS2322: Type 'unknown' is not assignable to type 'string | number'.
  Type 'unknown' is not assignable to type 'number'.

179               enumValue = this.config.enumValues[typeName].mappedValues[enumValue];
                  ~~~~~~~~~

packages/plugins/other/visitor-plugin-common/src/base-resolvers-visitor.ts:1161:63 - error TS2345: Argument of type '{ [valueName: string]: unknown; }' is not assignable to parameter of type '{ [valueName: string]: string | number; }'.
  Index signatures are incompatible.
    Type 'unknown' is not assignable to type 'string | number'.
      Type 'unknown' is not assignable to type 'number'.

1161           ? this.buildEnumResolversExplicitMappedValues(node, this.config.enumValues[rawTypeName].mappedValues)
                                                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

packages/plugins/other/visitor-plugin-common/src/base-types-visitor.ts:531:11 - error TS2322: Type 'unknown' is not assignable to type 'string | number'.
  Type 'unknown' is not assignable to type 'number'.

531           enumValue = this.config.enumValues[typeName].mappedValues[enumValue];
              ~~~~~~~~~

packages/plugins/typescript/typescript/src/visitor.ts:251:23 - error TS2322: Type 'unknown' is not assignable to type 'string | number'.
  Type 'unknown' is not assignable to type 'number'.

251                 const enumValue: string | number = getValueFromConfig(name) ?? name;
                          ~~~~~~~~~

packages/plugins/typescript/typescript/src/visitor.ts:271:21 - error TS2322: Type 'unknown' is not assignable to type 'string | number'.
  Type 'unknown' is not assignable to type 'number'.

271               const enumValue: string | number = valueFromConfig ?? i;
                        ~~~~~~~~~

packages/plugins/typescript/typescript/src/visitor.ts:301:21 - error TS2322: Type 'unknown' is not assignable to type 'string | number'.
  Type 'unknown' is not assignable to type 'number'.

301               const enumValue: string | number = getValueFromConfig(name) ?? name;
                        ~~~~~~~~~


Found 6 errors.
```

**Comment by cometkim** [comment]
I can reproduce https://github.com/cometkim/gatsby-plugin-typegen/issues/143 in gatsby-starter and confirmed this will fix the issue, I did test via manipulating node_modules directly

**Comment by cometkim** [comment]
This type is being used in the gatsby sharp plugin.

image

**Comment by ardatan** [comment]
@cometkim Thanks for the PR :) For changing enum values to `unknown` can be discussed inside another issue :)

File: packages/plugins/other/visitor-plugin-common/tests/enum-values.spec.ts [hunk 1]
```diff
@@ -147,4 +147,55 @@ describe('enumValues', () => {
...
+  });
+
+  it('should respect non-string enum values', () => {
+    const result = parseEnumValues({
+      schema: schemaWithNonStringEnumValues,
+      mapOrStr: {},
+      ignoreEnumValuesFromSchema: false,
+    });
+
+    expect(result).not.toEqual({
```
Comment (cometkim) [inline packages/plugins/other/visitor-plugin-common/tests/enum-values.spec.ts:185]: Oh, sorry about this. this test case doesn't make sense

Comment (ardatan) [reply packages/plugins/other/visitor-plugin-common/tests/enum-values.spec.ts:185]: @cometkim could you create another PR to fix that? :)

**Pull request merged**